pkg/providers/imagefamily/bootstrap/parser_helpers.go

```diff
@@ -563,33 +552,60 @@ func getGpuDriverVersion(vmSize string) string {
 // IsSgxEnabledSKU determines if an VM SKU has SGX driver support.
 func getIsSgxEnabledSKU(vmSize string) bool {
 	switch vmSize {
-	case "Standard_DC2s", "Standard_DC4s":
+	case vmSizeStandardDc2s, vmSizeStandardDc4s:
 		return true
 	}
 	return false
 }
 
 func getShouldConfigureHTTPProxy(httpProxyConfig *nbcontractv1.HTTPProxyConfig) bool {
+	if httpProxyConfig == nil {
+		return false
+	}
 	return httpProxyConfig.GetHttpProxy() != "" || httpProxyConfig.GetHttpsProxy() != ""
 }
 
 func getShouldConfigureHTTPProxyCA(httpProxyConfig *nbcontractv1.HTTPProxyConfig) bool {
+	if httpProxyConfig == nil {
+		return false
+	}
 	return httpProxyConfig.GetProxyTrustedCa() != ""
 }
 
 func getAzureEnvironmentFilepath(v *nbcontractv1.CustomCloudConfig) string {
-	if v.GetIsAksCustomCloud() {
+	if v != nil && v.GetIsAksCustomCloud() {
 		return fmt.Sprintf("/etc/kubernetes/%s.json", v.GetTargetEnvironment())
 	}
 	return ""
 }
 
-func getLBDisableOutboundSnat(lb *nbcontractv1.LoadBalancerConfig) bool {
-	if lb.GetLoadBalancerSku() != nbcontractv1.LoadBalancerConfig_STANDARD {
-		return false
+func getLinuxAdminUsername(username string) string {
+	if username == "" {
+		return defaultLinuxUser
 	}
-	if lb.DisableOutboundSnat == nil {
-		return false
+	return username
+}
+
+func getTargetEnvironment(v *nbcontractv1.CustomCloudConfig) string {
+	if v == nil {
```

Comment:
same here, nil check not needed, since will be caught by the empty string check with the getter below.